lib/src/mustachio/renderer_base.dart

```diff
@@ -183,10 +191,21 @@ abstract class RendererBase<T> {
         throw MustachioResolutionError(node.keySpan.message(
             "Failed to resolve '${e.name}' on ${e.contextType} while resolving "
             '$remainingNames as a property chain on any types in the context '
-            "chain: $contextChainString, after first resolving '${names.first}' "
-            'to a property on $T'));
+            "chain: $contextChainString, after first resolving '$firstName' to "
+            'a property on $T'));
       }
-    } else if (parent != null) {
+    }
+
+    if (this is SimpleRenderer && _invisibleGetters.contains(names.first)) {
+      var type = context.runtimeType;
+      throw MustachioResolutionError(node.keySpan.message(
```

Comment:
optional:  This seems like it should work, but there is no end to end test?

Reply:
OK, there were definitely bugs :P  added two tests after fixing the bugs.